src/hooks/usePets.js

```diff
@@ -1,22 +1,31 @@
 import { useEffect, useState } from "react";
 import { GetPets } from "../services/ApiService";
 
-const usePets = initialEndPoint => {
+const usePets = endPoint => {
   const [isLoading, setIsLoading] = useState(true);
   const [hasError, setHasError] = useState(false);
-
+  const [petItems, setPetItems] = useState([]);
   useEffect(() => {
-    GetPets()
-      .then(({ records }) => {})
+    GetPets(endPoint)
+      .then(response => {
+        const { records } = response;
+        records ? setPetItems([...records]) : setPetItems(response);
```

Comment:
pretty sure this could be simplified

```records || response```

I'm not sure though. I don't think the destructing the array buys us anything, but again not sure.

Reply:
done, though I think this will change anyway